core/src/main/java/org/ehcache/config/CacheConfigurationBuilder.java

```diff
@@ -59,9 +61,14 @@ private CacheConfigurationBuilder(final Expiry<? super K, ? super V> expiry, fin
     this.serviceConfigurations.addAll(serviceConfigurations);
   }
 
-  public CacheConfigurationBuilder<K, V> addServiceConfig(ServiceConfiguration<?> configuration) {
-    serviceConfigurations.add(configuration);
-    return this;
+  public CacheConfigurationBuilder<K, V> add(ServiceConfiguration<?> configuration) {
+    CacheConfigurationBuilder<K, V> otherBuilder = new CacheConfigurationBuilder<K, V>(expiry, classLoader, evictionPrioritizer, evictionVeto, resourcePools, serviceConfigurations);
```

Comment:
I would personally prefer to add a 'copy constructor' that takes a CacheConfigurationBuilder as parameter. Not a huge problem but I find that approach a bit simpler.